### ISSUE TITLE
[PHP] Wrong "latest version" published on PECL

### ISSUE THREAD

**Issue by mlocati** [protocolbuffers/protobuf#10619]
When we install the protobuf PHP extension, we usually install the latest version.
This is done by fetching the tarball available at https://pecl.php.net/get/protobuf (or at https://pecl.php.net/get/protobuf/stable)

The problem is that the protobuf versions published on the PECL archive have wrong dates (see https://pecl.php.net/package/protobuf):

![immagine](https://user-images.githubusercontent.com/928116/190968896-068b0042-cfba-498a-a4d1-6618e90d6819.png)

So, PECL thinks that the latest version is 3.18.3 instead of 3.21.6, and it serves that version.

**Comment by fowles** [comment]
I think that PECL is using strict "release date" and not actually looking at version numbers.  This might be fixed by simply re-pushing 3.21.6

**Comment by mlocati** [comment]
I absolutely don't know how PECL works, but sure, I agree that the first guess would be to fix the release dates

(and I'd do it for all the versions with wrong dates, that is the ones that *seems* to have been published on 2022-09-13 and on 2022-09-14)

**Comment by deannagarcia** [comment]
I agree with Matt here that the problem is that PECL is just sorting by release date and nothing else.

In our experience, PECL is extremely picky with how dates work. I believe this is a PECL issue and not a protobuf issue and would recommend you file a bug there with how versions are being dealt with. We will continue to have this issue with any patch release. 

Reuploading the package in PECL requires code changes and with it new tags, which I think will cause even more confusion.

**Comment by mlocati** [comment]
Maybe @cmb69 can give you a tip, @deannagarcia

**Comment by cmb69** [comment]
I don't think that the PECL Website even sorts by release date; instead, it looks like it shows releases in the order they have been published (might just sort by fmtime/fctime)). That works well for extensions with a "linear" release cycle, but not for extensions which do also support older branches with revisions. A work-around would be to publish in the desired order (e.g. 3.18.3, 3.19.5, 3.20.2, 3.21.6). A more sophisticated approach might be reasonable. Maybe someone wants to provide a PR to https://github.com/php/web-pecl. Maybe discussion on the pecl-dev mailing list would also be appropriate/useful.

**Comment by mlocati** [comment]
I'm closing this issue since it's related to PEAR/PECL in general, and not specifically to the protobuf PHP extension.

PS: I [wrote to the pecl-dev mailing list](https://marc.info/?m=168137915918907) about this problem.

**Comment by mlocati** [comment]
> I [wrote to the pecl-dev mailing list](https://marc.info/?m=168137915918907) about this problem.

Just an update: nobody replied to that message of mine in the pecl-dev mailing list, so [I suggested a fix](https://github.com/php/web-pecl/pull/92) directly in the repository of the pecl website.

**Comment by mlocati** [comment]
This issue has been resolved: see https://github.com/php/web-pecl/pull/92
